clients/python-wrapper/lakefs/client.py

```diff
@@ -106,6 +106,17 @@ def version(self) -> str:
         return self._server_conf.version
 
 
+def from_web_identity(code: str, state: str, redirect_uri: str, ttl_seconds: int = 3600, **kwargs) -> Client:
+    """
+    Authenticate against lakeFS using a web identity token
+    """
+    client = Client(**kwargs)
+    sts_requests = lakefs_sdk.StsAuthRequest(code=code, state=state, redirect_uri=redirect_uri, ttl_seconds=ttl_seconds)
+    with api_exception_handler():
+        auth_token = client.sdk_client.experimental_api.sts_login(sts_requests)
+        client.config.access_token = auth_token.token
+        return client
```

Comment:
Doesn't need to be under the exception handling
```suggestion
    
    client.config.access_token = auth_token.token
    return client
```